packages/loopring_v3/contracts/amm/libamm/AmmDepositProcess.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "../../lib/ERC20.sol";
+import "../../lib/MathUint96.sol";
+import "../../lib/TransferUtil.sol";
+import "./AmmAssetManagement.sol";
+import "./AmmData.sol";
+import "./AmmPoolToken.sol";
+import "./AmmStatus.sol";
+import "./AmmUtil.sol";
+import "./AmmWithdrawal.sol";
+
+
+/// @title AmmDepositProcess
+library AmmDepositProcess
+{
+    using AmmAssetManagement for AmmData.State;
+    using MathUint96         for uint96;
+    using TransferUtil       for address;
+
+    function processDeposit(
+        AmmData.State       storage S,
+        AmmData.Context     memory ctx,
+        AmmData.PoolDeposit memory poolDeposit
+        )
+        internal
+    {
+        require(poolDeposit.amounts.length == ctx.tokens.length, "INVALID_DEPOSIT_DATA");
+        for (uint i = 0; i < ctx.tokens.length; i++) {
+            address token = ctx.tokens[i].addr;
+            uint96 amount = poolDeposit.amounts[i];
```

Comment:
how about we check `if (amount > 0)` then do the deposit and update of `balancesL1`, otherwise just skip.

Reply:
Deposits/withdrawals transactions were already skipped when `amount == 0` but you're right some other things can be skipped as well. Updated.